src/main/java/core/basesyntax/service/impl/ParserImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dto.FruitDto;
+import core.basesyntax.service.Parser;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ParserImpl implements Parser {
+    private static final String CSV_FORMAT = "[b,s,r,p],[a-zA-Z]+,[0-9]+";
```

Comment:
character "," also matches
```suggestion
    private static final String CSV_FORMAT = "[bsrp],[a-zA-Z]+,[0-9]+";
```